src/app/clusters/occupancy-sensor-server/occupancy-sensor-server.h

```diff
@@ -17,8 +17,76 @@
 
 #pragma once
 
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
 #include <app/util/af-types.h>
 #include <app/util/basic-types.h>
+#include <app/util/config.h>
+#include <lib/core/DataModelTypes.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace OccupancySensing {
+
+/**
+ * Interface to help manage the Hold Time Limits of the Occupancy Sensing Cluster.
+ */
+class HoldTimeLimitsManager : public AttributeAccessInterface
+{
+public:
+    static constexpr size_t kOccupancySensingServerMaxEndpointCount =
+        MATTER_DM_OCCUPANCY_SENSING_CLUSTER_SERVER_ENDPOINT_COUNT + CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT;
```

Comment:
Do not use this pattern for memory allocation of endpoints as it over-allocates.

Reply:
Please check the new code, I refer to Electrical Energy Measurement Cluster's implementation for this, if I'm in the wrong way, help suggest a reference, thanks~